llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,383 @@
+
+        Examples:
+            store = GoogleVectorStore.create_corpus()
+            print(f"Created corpus with ID: {store.corpus_id})
+
+            store = GoogleVectorStore.create_corpus(
+                display_name="My first corpus"
+            )
+
+            store = GoogleVectorStore.create_corpus(
+                corpus_id="my-corpus-1",
+                display_name="My first corpus"
+            )
+
+        Args:
+            corpus_id: ID of the new corpus to be created. If not provided,
+                Google server will provide one for you.
+            display_name: Title of the corpus. If not provided, Google server
+                will provide one for you.
+
+        Returns:
+            An instance of the vector store that points to the specified corpus.
+        """
+        try:
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(
+            f"\n\nGoogleVectorStore.create_corpus(new_corpus_id={corpus_id}, new_display_name={display_name})"
+        )
+
+        client = genaix.build_semantic_retriever()
+        new_corpus_id = corpus_id or str(uuid.uuid4())
+        new_corpus = genaix.create_corpus(
+            corpus_id=new_corpus_id, display_name=display_name, client=client
+        )
+        name = genaix.EntityName.from_str(new_corpus.name)
+        return cls(corpus_id=name.corpus_id, client=client)
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "GoogleVectorStore"
+
+    @property
+    def client(self) -> Any:
+        return self._client
+
+    def add(self, nodes: List[BaseNode]) -> List[str]:
+        """Add nodes with embedding to vector store.
+
+        If a node has a source node, the source node's ID will be used to create
+        a document. Otherwise, a default document for that corpus will be used
+        to house the node.
+
+        Furthermore, if the source node has a metadata field "file_name", it
+        will be used as the title of the document. If the source node has no
+        such field, Google server will assign a title to the document.
+
+        Example:
+            store = GoogleVectorStore.from_corpus(corpus_id="123")
+            store.add([
+                TextNode(
+                    text="Hello, my darling",
+                    relationships={
+                        NodeRelationship.SOURCE: RelatedNodeInfo(
+                            node_id="doc-456",
+                            metadata={"file_name": "Title for doc-456"},
+                        )
+                    },
+                ),
+                TextNode(
+                    text="Goodbye, my baby",
+                    relationships={
+                        NodeRelationship.SOURCE: RelatedNodeInfo(
+                            node_id="doc-456",
+                            metadata={"file_name": "Title for doc-456"},
```

Comment:
Do they have to specify the file_name every time or just for the first node with that doc ID?

Reply:
Implementation-wise, just the first time. So, if there are two nodes with the same source node ID but different file_name, only the first one will stick.

Having said that, typically these metadata are generated by LlamaIndex framework. So, this inconsistency shouldn't be there.